tensorboard/webapp/metrics/store/metrics_reducers.ts

```diff
@@ -325,9 +310,21 @@ const reducer = createReducer(
       state.pinnedCardToOriginal,
       state.cardStepIndex
     );
+
+    const hydratedSmoothing = hydratedState.metrics.smoothing;
+    const newSmoothing =
+      Number.isFinite(hydratedSmoothing) && hydratedSmoothing !== null
+        ? Math.max(0, Math.min(0.999, hydratedSmoothing))
+        : state.settings.scalarSmoothing;
+    const newSettings = {
+      ...state.settings,
+      scalarSmoothing: newSmoothing,
+    };
```

Comment:
Maybe my suggestion leans a bit towards overoptimizing, but in the case when `newSmoothing` has not changed and we can reuse the old settings, should we return `state.settings` instead of a new object?